### PR TITLE
refactor(link): add button alternative and typo prop LUM-11250

### PR DESCRIPTION
**Pull request by pidupuis**
# General summary

- A `Link` component without an `href` would result in a button looking like a link instead of an anchor.
Capture d’écran de 2020-11-09 17-30-00
- Added `typography` prop to `Link` component.
Capture d’écran de 2020-11-09 17-32-39

# Check list

-   [x] (if has style) Add `need: review-integration` label
-   [ ] (if has textual documentation) Add `need: review-doc` label
-   [x] (if has code) Add `need: review-frontend` label
-   [x] (if has react) Check through the [react dev check list]
-   [x] (if fix or feature) Add `need: test` label

Check out the [contribution guide] for general guidelines for submissions to the design system.

[contribution guide]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md
[react dev check list]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md#react-developpment-check-list

### CODE REVIEW

File: packages/lumx-react/src/components/link/Link.tsx
Comment (Mcourt) [inline packages/lumx-react/src/components/link/Link.tsx:59]: image

**Review by Mcourt — APPROVED** [review]
LGTM (minus some DS specificities I might have overlooked, it might be a good idea to also wait for @gcornut's approval :) )

File: packages/lumx-react/src/components/link/Link.tsx
Comment (gcornut) [inline packages/lumx-react/src/components/link/Link.tsx:67]: handleBasicClasses({ prefix: 'lumx', typography }),

**Pull request merged**
